### DISCUSSION TITLE
Why should I return ORJSONResponse

### DISCUSSION THREAD

**Discussion by idan3** [fastapi/fastapi#11265] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Description

In the documentation, the example is:

```
@app.get("/items/", response_class=ORJSONResponse)
async def read_items():
    return ORJSONResponse([{"item_id": "Foo"}])
```
I do not understand why I need to do `ORJSONResponse([{"item_id": "Foo"}])`? This works as well:

```
@app.get("/items/", response_class=ORJSONResponse)
async def read_items():
    return [{"item_id": "Foo"}]   # Or: return MyResponse("hello")
```
It also works with class objects and is more readable and still (If I understand correctly) orjson is the one doing the serialization.

### Operating System

macOS

### FastAPI Version

0.109.2

### Pydantic Version

2.6.1

### Python Version

Python 3.10.13

**Comment by YuriiMotov** [comment, marked as answer]
It's for optimization purposes. This can be useful if your return data is big and you are sure that it's serializable with JSON.

As it's said in the documentation, by returning `ORJSONResponse([{"item_id": "Foo"}])` you can 

> avoid the extra overhead that FastAPI would have by passing your return content through the `jsonable_encoder` before passing it to the response class.

https://fastapi.tiangolo.com/advanced/custom-response/#use-orjsonresponse

`ORJSONResponse` is inherited from the `Response` class. So, in this case you will save some execution time by not passing your return content through the `jsonable_encoder`. If you just return `[{"item_id": "Foo"}]`, FastAPI will pass it through the `jsonable_encoder` and then validate.
